pulsar/consumer_partition.go

```diff
@@ -138,15 +138,14 @@ type partitionConsumer struct {
 	// the size of the queue channel for buffering messages
 	queueSize       int32
 	queueCh         chan []*message
-	startMessageID  trackingMessageID
+	startMessageID  atomicMessageID
```

Comment:
Could `atomic.Value{}` instead of this?

Reply:
> Could `atomic.Value{}` instead of this?

I think atomicMessageID is better than `atomic.Value{}` because it's more simple and clearly.

For example, if we use `atomic.Value` as the startMessageID type, the original using of `startMessageID` will be like this.
``` golang
// original
return pc.startMessageID.greater(msgID.messageID)

// atomicMessageID
return pc.startMessageID.get().greater(msgID.messageID)

// atomic.Value{}
return pc.startMessageID.Load().(trackingMessageID).greater(msgID.messageID)
```
`atomic.Value{}` will need one more time type assertion.

But `atomicMessageID` may need a better name and declear position. Do you have more idea? Thanks.

Reply:
LGTM